### DISCUSSION TITLE
What is the latest version of hcxdumptool that supports the --filterlist_ap= parameter?

### DISCUSSION THREAD

**Discussion by mickhaorex** [ZerBea/hcxdumptool#457] [Q&A]
In the latest versions of hcxdumptool I have to create a filter.bpf file to filter APs. This is very inconvenient in terms of removing or adding MAC addresses. So I have a question to the respected developer and community, what is the latest version of hcxdumptool that supports the --filterlist_ap= parameter?
It would be possible to simply add or remove MAC addresses to the list without compiling a separate file.

**Comment by ZerBea** [comment]
The latest version that use WIRELESS EXTENSIONS and the old filter method is 6.2.9

changelog:
```
06.04.2023
==========
release v6.2.9
several bug fixes

This is the last version:
that use WIRELESS EXTENSIONS
that use ETHTOOL to get/set virtual MAC address
that use old style status output
that use soft coded filter lists
that use msec timestamp
that use crypto stuff
that use server/client mode to display status

Next version will go back to the roots:
set focus on WPA PSK (WPA1, WPA2, WPA2 key version 3)
set bandwidth to 20MHz to increase range
set bitrate to lowest values to increase range
use active monitor mode
use NL80211 stack
use RTNETLINK
band a, b, c, d, e support
use NMEA messages:
 $GPRMC: Position, velocity, time and date
 $GPGGA: Position, orthometric height, fix related data, time
 $GPWPL: Position and MAC AP
 $GPTXT: ESSID in HEX ASCII
remove options that slow hcxdumptool down
```

I do not recommend to use this version on Linux kernel > 6.0 because WIRELESS EXTENSIONS are deprecated and WiFi 7 drivers do not support them any longer.

BTW:
`It would be possible to simply add or remove MAC addresses to the list without compiling a separate file.`
That is exactly the purpose of a bash script.

It is very simple to code a simple bash script to generate and compile a BPF, because hcxdumptool/hcxlabtool is designed to run inside scripts. 

A basic example that take a BSSID and compile the filter:
```
#!/bin/bash

hcxdumptool --bpfc="wlan addr3 ffffffffffff or wlan addr3 $1" > $2
```

The same applies to hcxpcapngtool to filter a BSSID by a bash script:
https://github.com/ZerBea/hcxtools/issues/336#issuecomment-2136618194

Usage: filter BSSID FILTERNAME
`./filter 112233445566 target.bpf`

It doesn't make sense to integrate it to hcxdumptool because bash scripts can do this much better. And they are much more flexible.
